drake/systems/plants/collision/bullet_model.cc

```diff
@@ -637,17 +639,15 @@ bool BulletModel::closestPointsAllToAll(
     const std::vector<ElementId>& ids_to_check, const bool use_margins,
     std::vector<PointPair>& closest_points) {
   std::vector<ElementIdPair> id_pairs;
-  for (auto idA_iter = ids_to_check.begin(); idA_iter != ids_to_check.end();
-       ++idA_iter) {
-    auto elementA_iter = elements.find(*idA_iter);
-    if (elementA_iter != elements.end()) {
-      for (auto idB_iter = idA_iter + 1; idB_iter != ids_to_check.end();
-           ++idB_iter) {
-        auto elementB_iter = elements.find(*idB_iter);
-        if (elementB_iter != elements.end()) {
-          if (elements[*idA_iter]->collidesWith(elements[*idB_iter].get())) {
-            id_pairs.push_back(std::make_pair(*idA_iter, *idB_iter));
-          }
+  for (int i = 0; i < ids_to_check.size(); ++i) {
```

Comment:
In the past I was told not to use single-character variable names because that would prevent quick search-and-replace or multi-cursor highlighting in a text editor. For example, if a variable is named `i`, I cannot easily search for all references to it within a function, since the letter `i` will likely be used in numerous other ways beyond referring to the variable. The style recommendation was to double-up the letters. For example, instead of `i`, `j`, `k`, etc., use double letters like `ii`, `jj`, `kk`, etc. By doubling the character, the probability of encountering an instance of it that did not refer to the actual variable is greatly reduced, though not eliminated.

Here's an illustration using the for-loop example given above. Below is the multi-cursor selection when using a single character `i` as the variable name. Notice it selects a bunch of "i" characters in the comment above the code.

![single_char_var](https://cloud.githubusercontent.com/assets/1388098/14068731/02152992-f45b-11e5-8c9c-2bba9570947e.png)

Below is the multi-cursor selection when `ii` is used instead. Notice only instances of the variable are highlighted.

![double_char_var](https://cloud.githubusercontent.com/assets/1388098/14068751/5dd496a0-f45b-11e5-8973-30b64ffe7c95.png)

I actually don't feel too strongly about this since I'm sure most advanced IDEs will be able to highlight just the variable regardless of its name. Just wanted to explain why certain style guides may recommend avoiding single-character variable names.